experimental/dds/tree2/src/feature-libraries/editable-tree-2/lazyField.ts

```diff
@@ -254,26 +256,122 @@ export class LazySequence<TTypes extends AllowedTypes>
 		const fieldEditor = this.context.editor.sequenceField(fieldPath);
 		return fieldEditor;
 	}
 
-	public replaceRange(
-		index: number,
-		count: number,
-		newContent: Iterable<FlexibleNodeContent<TTypes>>,
-	): void {
+	public insertAt(index: number, value: FlexibleNodeContent<TTypes>[]): void {
 		const fieldEditor = this.sequenceEditor();
-		const content = this.normalizeNewContent([...newContent]);
-
-		fieldEditor.delete(index, count);
+		const content = this.normalizeNewContent(Array.isArray(value) ? value : [value]);
+		assertValidIndex(index, this, true);
 		fieldEditor.insert(index, content);
 	}
 
-	public get asArray(): readonly UnboxNodeUnion<TTypes>[] {
-		return this.map((x) => x);
+	public insertAtStart(value: FlexibleNodeContent<TTypes>[]): void {
+		this.insertAt(0, value);
+	}
+
+	public insertAtEnd(value: FlexibleNodeContent<TTypes>[]): void {
+		this.insertAt(this.length, value);
+	}
+
+	public removeAt(index: number): void {
+		const fieldEditor = this.sequenceEditor();
+		fieldEditor.delete(index, 1);
+	}
+
+	public removeRange(start?: number, end?: number): void {
+		const fieldEditor = this.sequenceEditor();
+		const { length } = this;
+		const removeStart = start ?? 0;
+		const removeEnd = Math.min(length, end ?? length);
+		assertValidRangeIndices(removeStart, removeEnd, this);
+		fieldEditor.delete(removeStart, removeEnd - removeStart);
+	}
+
+	public moveToStart(sourceStart: number, sourceEnd: number): void;
+	public moveToStart<TTypesSource extends AllowedTypes>(
+		sourceStart: number,
+		sourceEnd: number,
+		source: Sequence<CheckTypesOverlap<TTypesSource, TTypes>>,
+	): void;
+	public moveToStart<TTypesSource extends AllowedTypes>(
+		sourceStart: number,
+		sourceEnd: number,
+		source?: Sequence<CheckTypesOverlap<TTypesSource, TTypes>>,
+	): void {
+		this.moveToIndexI(0, sourceStart, sourceEnd, source);
+	}
+
+	public moveToEnd(sourceStart: number, sourceEnd: number): void;
+	public moveToEnd<TTypesSource extends AllowedTypes>(
+		sourceStart: number,
+		sourceEnd: number,
+		source: Sequence<CheckTypesOverlap<TTypesSource, TTypes>>,
+	): void;
+	public moveToEnd<TTypesSource extends AllowedTypes>(
+		sourceStart: number,
+		sourceEnd: number,
+		source?: Sequence<CheckTypesOverlap<TTypesSource, TTypes>>,
+	): void {
+		this.moveToIndexI(this.length, sourceStart, sourceEnd, source);
+	}
+
+	public moveToIndex(index: number, sourceStart: number, sourceEnd: number): void;
+	public moveToIndex<TTypesSource extends AllowedTypes>(
+		index: number,
+		sourceStart: number,
+		sourceEnd: number,
+		source: Sequence<CheckTypesOverlap<TTypesSource, TTypes>>,
+	): void;
+	public moveToIndex<TTypesSource extends AllowedTypes>(
+		index: number,
+		sourceStart: number,
+		sourceEnd: number,
+		source?: Sequence<CheckTypesOverlap<TTypesSource, TTypes>>,
+	): void {
+		this.moveToIndexI(index, sourceStart, sourceEnd, source);
+	}
+
+	private moveToIndexI<TTypesSource extends AllowedTypes>(
```

Comment:
I think we've generally been using `_` prefixes for private counterparts to public APIs elsewhere in the repo. Has this package been using the `I` postfix instead?

Reply:
@noencke I defer to you here. What have we been doing?

Reply:
We don't have a convention in this package. I didn't know about `_`. Without thinking about it super hard, I like `_`, personally.

Reply:
I will make it so.

Reply:
I was just curious. No need to block the PR on this. Though longer term a single convention is always nice :)